autogpt/core/planning/base.py

```diff
@@ -1,5 +1,40 @@
 import abc
 
-
 class Planner(abc.ABC):
-    pass
+    """Build prompts based on inputs, can potentially store and retrieve planning state from the workspace"""
+
+    def __init__(config, logger, plugin_manager, workspace, *arg, **kwargs):
```

Comment:
Here we want our methods to be abstract, including the init.  W'ere mainly here to define method signatures.

```suggestion
    @abc.abstractmethod
    def __init__(
        configuration: Configuration, 
        logger: Logger, 
        plugin_manager: PluginManager, 
        workspace: Workspace
    ) -> None:
        ...
```